runtime/parser/parse_model.go

```diff
@@ -17,17 +17,23 @@ import (
 
 // ModelYAML is the raw structure of a Model resource defined in YAML (does not include common fields)
 type ModelYAML struct {
-	commonYAML            `yaml:",inline" mapstructure:",squash"` // Only to avoid loading common fields into InputProperties
-	Refresh               *ScheduleYAML                           `yaml:"refresh"`
-	Timeout               string                                  `yaml:"timeout"`
-	Incremental           bool                                    `yaml:"incremental"`
-	ChangeMode            string                                  `yaml:"change_mode"`
-	State                 *DataYAML                               `yaml:"state"`
-	Partitions            *DataYAML                               `yaml:"partitions"`
-	Splits                *DataYAML                               `yaml:"splits"` // Deprecated: use "partitions" instead
-	PartitionsWatermark   string                                  `yaml:"partitions_watermark"`
-	PartitionsConcurrency uint                                    `yaml:"partitions_concurrency"`
-	InputProperties       map[string]any                          `yaml:",inline" mapstructure:",remain"`
+	commonYAML  `yaml:",inline" mapstructure:",squash"` // Only to avoid loading common fields into InputProperties
+	Refresh     *ScheduleYAML                           `yaml:"refresh"`
+	Timeout     string                                  `yaml:"timeout"`
+	Incremental bool                                    `yaml:"incremental"`
+	ChangeMode  string                                  `yaml:"change_mode"`
+	State       *DataYAML                               `yaml:"state"`
+	Partitions  *DataYAML                               `yaml:"partitions"`
+	// TODO: Consider changing to (data_tests, assertions) in the future
+	Tests []struct {
+		Name        string `yaml:"name"`
+		Description string `yaml:"description"`
```

Comment:
Open to input, but keep only `name` and remove `description`? The names can anyway be long if you want, and additional info can be put in YAML comments.

Reply:
Fair enough, I can drop description